gazebo_plugins/src/gazebo_ros_wheel_slip.cpp

```diff
@@ -110,10 +129,21 @@ void GazeboRosWheelSlip::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
   // Declare parameters after adding callback so that callback will trigger immediately.
   // Set negative values by default, which are ignored by the callback.
   // This approach allows values specified in a launch file to override the SDF/URDF values.
-  impl_->ros_node_->declare_parameter("slip_compliance_unitless_lateral", slip_lateral_default);
-  impl_->ros_node_->declare_parameter(
-    "slip_compliance_unitless_longitudinal",
-    slip_longitudinal_default);
+  for (auto &wheel_parameter : impl_->mapSlipLateralDefault)
+  {
+    impl_->ros_node_->declare_parameter(
+        "slip_compliance_unitless_lateral/" + wheel_parameter.first,
```

Comment:
My only concern with this PR is that it creates a hard API break wrt the names of the ROS parameters. It would be nice if users didn't have to update their code, and could continue relying on the old behavior. Maybe we can keep the original parameters for updating slip values for all the wheels in addition to the new, per-wheel, parameters? I'm open to other ideas.

Reply:
Yup that makes sense. I can keep the original parameters as well which apply to all the wheels.